src/fairseq2/recipes/hg/asr_eval.py

```diff
@@ -232,6 +271,79 @@ def load_wav2vec2_asr_evaluator(
         gang=gang,
         data_reader=pipeline_reader,
         wall_watch=wall_watch,
-        preprocessor=seq2seq_preprocessor,
-        postprocessor=lambda x, y: postprocesser(x, y, tokenizer),
+        preprocessor=evaluator_preprocessor,
+        postprocessor=lambda x, y: evaluator_postprocesser(x, y, tokenizer),
+    )
+
+
+class HGModelWrapper:
+    def __init__(self, model: WhisperForConditionalGeneration):
+        self.model = model
+
+    def __call__(self, batch: SequenceBatch) -> Any:
+        return self.model.generate(batch.seqs)
+
+
+def load_hg_asr_evaluator(
+    config: AsrEvalConfig, output_dir: Path
+) -> HFEvaluator[Seq2SeqBatch]:
+    """
+    Load the evaluator used for downstream evaluation of the whisper model
+
+    Args:
+        config (HFEvalConfig): The configuration for the evaluation.
+        output_dir (Path): The output directory to store the evaluation results.
+
+    Returns:
+        HFEvaluator: Evaluation process.
+    """
+    if not isinstance(config, AsrEvalConfig):
+        raise ValueError(f"Expect AsrEvalConfig, get {type(config)}")
+
+    gang = setup_root_gang(log)
+
+    if gang.rank == 0:
+        init_device = gang.device
+    else:
+        init_device = META
+
+    processor = WhisperProcessor.from_pretrained(config.model_name)
+    model = WhisperForConditionalGeneration.from_pretrained(config.model_name).to(
+        init_device
+    )
+
+    ds_builder = load_dataset_builder(config.dataset_name)
+    ds = prepare_dataset(
+        config,
+        lambda x: {
+            "audio": processor(
+                x["audio"],
+                sampling_rate=ds_builder.info.features["audio"].sampling_rate,
+                return_tensors="pt",
+            ).input_features.squeeze(0)
+        },
+    )
+
+    pipeline_reader = create_hf_reader(
+        dataset=ds,
+        gang=gang,
+        converter=lambda x: to_batch(x, "whisper", init_device),
+        batching=StaticBatching(config.max_num_elements),
+        num_prefetch=config.num_prefetch,
+        max_seq_len=config.max_audio_len,
+    )
+
+    wall_watch = Stopwatch(start=True, device=init_device)
+
+    return HFEvaluator[Seq2SeqBatch](
+        model=cast(Model, HGModelWrapper(model)),
```

Comment:
See my other comment. 

This forced casting is artificial IMHO - The better option is to create a [model loader](https://github.com/facebookresearch/fairseq2/blob/main/src/fairseq2/models/loader.py#L70) that creates a HG model out of its name. But let's make it in a separate PR.